components/browser/engine-system/src/main/java/mozilla/components/browser/engine/system/SystemEngineView.kt

```diff
@@ -273,6 +275,42 @@ class SystemEngineView @JvmOverloads constructor(
                 }
             }
         }
+
+        override fun onReceivedHttpAuthRequest(view: WebView, handler: HttpAuthHandler, host: String, realm: String) {
+            val session = session ?: return handler.cancel()
+
+            val formattedUrl = session.currentUrl.toUri().let { uri ->
+                (uri.scheme ?: "http") + "://" + (uri.host ?: host)
+            }
+
+            // Trim obnoxiously long realms.
+            val trimmedRealm = if (realm.length > MAX_REALM_LENGTH) {
+                realm.substring(0, MAX_REALM_LENGTH) + "\u2026"
+            } else {
+                realm
+            }
+
+            val message = if (trimmedRealm.isEmpty()) {
+                context.getString(R.string.mozac_browser_engine_system_auth_no_realm_message, formattedUrl)
+            } else {
+                context.getString(R.string.mozac_browser_engine_system_auth_message, trimmedRealm, formattedUrl)
+            }
+
+            session.notifyObservers {
+                onPromptRequest(
+                    PromptRequest.Authentication(
+                        "",
+                        message,
+                        "",
```

Comment:
You could get the user and password from `Webview` by using [WebViewDatabase.getHttpAuthUsernamePassword](https://developer.android.com/reference/android/webkit/WebViewDatabase.html#getHttpAuthUsernamePassword(java.lang.String,%20java.lang.String)) or [webView.getHttpAuthUsernamePassword](https://developer.android.com/reference/android/webkit/WebView#getHttpAuthUsernamePassword(java.lang.String,%2520java.lang.String)).

```kotlin
        private fun WebView.getAuthCredentials(host: String, realm: String): Pair<String, String> {

            val credentials = if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
                val webViewDB = WebViewDatabase.getInstance(context)
                webViewDB.getHttpAuthUsernamePassword(host, realm)
            } else {
                this.getHttpAuthUsernamePassword(host, realm)
            }

            var credentialsPair = "" to ""

            if (!credentials.isNullOrEmpty() && credentials.size == 2) {

                val user = if (credentials[0] != null) credentials[0] else ""
                val pass = if (credentials[1] != null) credentials[1] else ""

                credentialsPair = user to pass
            }
            return credentialsPair
        }
```